### PR TITLE
RPP Test suite : Modifications to support python 2 

### PR DESCRIPTION
**Pull request by HazarathKumarM**
- Adds changes to support tests for all the versions of python

### CODE REVIEW

File: utilities/test_suite/HIP/runMiscTests.py [hunk 1]
```diff
@@ -74,25 +74,25 @@ def generate_performance_reports(RESULTS_DIR):
     print(dfPrint_noIndices)
 
 def run_unit_test_cmd(numDims, case, numRuns, testType, toggle, batchSize, outFilePath, additionalArg):
-    print(f"./Tensor_misc_hip {case} {testType} {toggle} {numDims} {batchSize} {numRuns} {additionalArg}")
-    result = subprocess.run([buildFolderPath + "/build/Tensor_misc_hip", str(case), str(testType), str(toggle), str(numDims), str(batchSize), str(numRuns), str(additionalArg), outFilePath, scriptPath], stdout=subprocess.PIPE)    # nosec
-    print(result.stdout.decode())
+    print("./Tensor_misc_hip {} {} {} {} {} {} {}".format(case, testType, toggle, numDims, batchSize, numRuns, additionalArg))
+    result = subprocess.Popen([buildFolderPath + "/build/Tensor_misc_hip", str(case), str(testType), str(toggle), str(numDims), str(batchSize), str(numRuns), str(additionalArg), outFilePath, scriptPath], stdout=subprocess.PIPE)    # nosec
```
Comment (r-abishek) [inline utilities/test_suite/HIP/runMiscTests.py:78]: Why the change from subprocess.run to subprocess.Popen here?

File: utilities/test_suite/HIP/runMiscTests.py [hunk 2]
```diff
@@ -206,8 +206,8 @@ def rpp_test_suite_parser_and_validator():
 os.chdir(buildFolderPath + "/build")
 
 # Run cmake and make commands
-subprocess.run(["cmake", scriptPath], cwd=".")   # nosec
-subprocess.run(["make", "-j16"], cwd=".")    # nosec
+subprocess.call(["cmake", scriptPath], cwd=".")   # nosec
+subprocess.call(["make", "-j16"], cwd=".")    # nosec
```
Comment (r-abishek) [inline utilities/test_suite/HIP/runMiscTests.py:210]: What's the difference in changing from run to call? vs the Popen above?

File: utilities/test_suite/HIP/runTests.py [hunk 3]
```diff
@@ -271,17 +275,17 @@ def rpp_test_suite_parser_and_validator():
...
 # List of cases supported
 supportedCaseList = ['0', '1', '2', '4', '8', '13', '20', '21', '23', '29', '30', '31', '32', '33', '34', '35', '36', '37', '38', '39', '45', '46', '54', '61', '63', '65', '68', '70', '79', '80', '82', '83', '84', '85', '86', '87', '88', '89', '90', '91', '92']
 
 # Create folders based on testType and profilingOption
 if testType == 1 and profilingOption == "YES":
-    os.makedirs(f"{dstPath}/Tensor_PKD3")
-    os.makedirs(f"{dstPath}/Tensor_PLN1")
-    os.makedirs(f"{dstPath}/Tensor_PLN3")
+    os.makedirs("{}/Tensor_PKD3".format(dstPath))
+    os.makedirs("{}/Tensor_PLN1".format(dstPath))
```
Comment (r-abishek) [inline utilities/test_suite/HIP/runTests.py:287]: Doesn't string concatenation work for all Python versions? Like using (dstPath + "/Tensor_PKD3")? 
Or is the .format() useful somewhere else?

**Review by r-abishek — COMMENTED** [review]
@HazarathKumarM Please address comments

File: utilities/test_suite/HIP/runMiscTests.py (hunk 1 above)
Comment (HazarathKumarM) [reply utilities/test_suite/HIP/runMiscTests.py:78]: .run methos is only introduced in python3.5 , so we need to change it to Popen to support running in python2

File: utilities/test_suite/HIP/runMiscTests.py (hunk 2 above)
Comment (HazarathKumarM) [reply utilities/test_suite/HIP/runMiscTests.py:211]: .call returns either 0 or 1 as output, in the above case we need to print the log in terminal and as well as to the log file , we need stdout to be returned from the subprocess, Popen return that log

File: utilities/test_suite/HIP/runTests.py (hunk 3 above)
Comment (HazarathKumarM) [reply utilities/test_suite/HIP/runTests.py]: changed to concatenation

**Comment by HazarathKumarM** [comment]
@r-abishek  currently there are no deprecation warnings for .call() and .Popen() methods

**Review by r-abishek — APPROVED** [review]
@HazarathKumarM 
Test confirmation on python 2.7, 3.5, 3.10
